### PR TITLE
AI Client: update and export SuggestionsEventSource library

### LINKED ISSUE #31935: AI Client: Update and integrate SuggestionsEventSource lib
The task involves updating the SuggestionsEventSource as defined in our package, in line with the latest modifications we've made in the Jetpack plugin.
Following this, we will update the Jetpack plugin to consume the SuggestionsEventSource from the AI Client package.

Ensure to add essential but helpful documentation.

### PR DESCRIPTION
**Pull request by retrofox**
This PR involves the following updates:

1. It refines and exports the `SuggestionsEventSource` class from the AI Client package.
2. It introduces the `SuggestionsEventSource` into the Jetpack plugin by importing and implementing it.
3. The duplicated code is streamlined for better readability and efficiency.

From the perspective of the SuggestionsEventSource, a few non-breaking compatibility changes have been made:

- The `url` parameter is now optional, meaning passing the URL when utilizing the class is no longer mandatory.
- The class can now handle the 'fromCache' option.
- It also accepts the 'feature' option.

Take a look a the [SuggestionsEventSource doc page](https://github.com/Automattic/jetpack/blob/89d002b6aec2ebb2049e6b22ed78cb903cd52aaf/projects/js-packages/ai-client/src/suggestions-event-source/Readme.md) for more detail.

Fixes partially https://github.com/Automattic/jetpack/issues/31936
Fixes https://github.com/Automattic/jetpack/issues/31935

## Proposed changes:

* AI Client: update and export SuggestionsEventSource library

### Other information:

- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

The best way to test this PR is simply using the AI Assistant block of the Jetpack plugin.

* Go to the block editor
* Create an AI Assistant block, or use the AI Extension tool
* Request changes.
* Confirm it works as expected.
* You can take a look at the debug logs

```
localStorage.setItem( 'debug', 'jetpack-*' )
```

The following screenshot shows how the lib defined the `url`, and also sets the `ai-assistant-experimental` feature.

Screenshot 2023-07-18 at 15 12 51

Use the AI Playground to test it:

Screenshot 2023-07-18 at 15 13 08

You can take a look at the request payload, too.

image

**`Done` log**

image

### Testing error response

You can hardcode the jetpack-ai-query file to force sending errors from the server side when responding to the request to be able to see how the app handles it—for instance, trying to send `422`, `429`, `500` and `503` status code.

This is a test for error code `500`:

Screenshot 2023-07-18 at 17 32 02

* Also, you can confirm the `error_unclear_prompt` (JETPACK_AI_ERROR) is properly handled.

image

image

### CODE REVIEW

**Review by renatoagds — APPROVED** [review]
LGTM. Works as expected based on my tests.

CleanShot 2023-07-19 at 17 56 11

CleanShot 2023-07-19 at 17 55 43

CleanShot 2023-07-19 at 17 54 53

File: projects/js-packages/ai-client/src/suggestions-event-source/index.ts [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	checkForUnclearPrompt() {
+		if ( this.isPromptClear ) {
+			return;
+		}
+
+		/*
+		 * Sometimes the first token of the message is not received,
+		 * so we check only for JETPACK_AI_ERROR, ignoring:
+		 * - the double underscores (italic markdown)
+		 * - the doouble asterisks (bold markdown)
```
Comment (dhasilva) [inline projects/js-packages/ai-client/src/suggestions-event-source/index.ts:168]:
```suggestion
		 * - the double asterisks (bold markdown)
```

Comment (retrofox) [reply projects/js-packages/ai-client/src/suggestions-event-source/index.ts:168]: Thanks. Let's considering to fix it in a follow-up 🤚

Comment (retrofox) [reply projects/js-packages/ai-client/src/suggestions-event-source/index.ts:168]: ddc4ae5e0be005f329e2e9791fbdc260560159e6

**Pull request merged**